### ISSUE TITLE
Symbol's function definition is void: org-projectile-project-todo-entry

### ISSUE THREAD

**Issue by laidlerc** [syl20bnr/spacemacs#11864]
**Description**
Trying to setup my own capture-templates for org-projectile but I'm receiving this error
This is the line in my .spacemacs
  (push (org-projectile-project-todo-entry :capture-template "* TODO %? %A\n" :capture-heading "Create linked org-projectile TODO" :capture-character "l") org-capture-templates)

Basically org-projectile-project-todo-entry is not found even though it is documented in org-projectile

**Reproduction** 
Only can be reproduced when that's in the .spacemacs file.

OS: linux
Emacs: 26.41
Spacemacs branch: develop 
Distribution: spacemacs
Editing style: vim
Completion: helm

**Comment by dangom** [comment]
Wrap your call within an eval-after-load:
```elisp
(with-eval-after-load 'org-projectile
    ;; your code here
)
```

**Comment by dangom** [comment]
in which case you can close the issue

**Comment by kitaev-chen** [comment]
> Wrap your call within an eval-after-load:
> 
> ```emacs-lisp
> (with-eval-after-load 'org-projectile
>     ;; your code here
> )
> ```

I also have this problem but I have already followed the document like this:
http://develop.spacemacs.org/layers/+emacs/org/README.html

```
(with-eval-after-load 'org-agenda
  (require 'org-projectile)
  (mapcar '(lambda (file)
                 (when (file-exists-p file)
                   (push file org-agenda-files)))
          (org-projectile-todo-files)))
```

so how to fix it? (I also see some discussion on the bug fixed on org-projectile #10434, but I don't know wether it will affect the emacs update if I cherrypick that bug fixed branch)

**Comment by smile13241324** [comment]
@kitaev-chen I think you need to check the documentation from the layer again.
There it says to wait for org to be loaded before running your commands not to wait for `org-agenda` which would not wait long enough.

See below:
```elisp
(with-eval-after-load 'org
  ;; here goes your Org config :)
  ;; ....
  )
```
In addition never require in your user config, this messes with spacemacs intricate loading procedures and can cause a lot of different bugs. Without the require part running your code within the properly guarded block should work flawlessly.

As a reference here is my org config:
```elisp
  ;; Configure org mode
  (with-eval-after-load 'org
    (setq org-agenda-files (quote ("~/Documents/GTD/CalendarActionList.org" "~/Documents/GTD/NextActionList.org" "~/Documents/GTD/ProjectList.org")))
    (setq org-refile-targets (quote (("~/Documents/GTD/CalendarActionList.org" :maxlevel . 1)
                                     ("~/Documents/GTD/NextActionList.org" :maxlevel . 1)
                                     ("~/Documents/GTD/ProjectList.org" :maxlevel . 1)
                                     ("~/Documents/GTD/Reference.org" :maxlevel . 1)
                                     ("~/Documents/GTD/SomedayMaybe.org" :maxlevel . 1)
                                     ("~/Documents/GTD/WaitingFor.org" :maxlevel . 1))))
    (setq org-enforce-todo-dependencies t)
    (setq org-enforce-todo-checkbox-dependencies t)
    (setq org-brain-path "~/Documents/GTD")
    (setq org-default-notes-file "~/Documents/GTD/Inbox.org")
    (setq org-archive-location "~/Documents/GTD/Reference.org::* Archived from %s")
    (setq spaceline-org-clock-p t)
    (setq org-capture-templates
          '(("i" "Input" entry (file "")
             "* %? %i")
            ("r" "General reference" entry (file+headline "~/Documents/GTD/Reference.org" "General")
             "* %? %i")))
    (setq org-agenda-skip-scheduled-if-done t)
    (setq org-agenda-skip-deadline-if-done t)
    (setq org-agenda-skip-timestamp-if-done t)
    (setq org-agenda-todo-list-sublevels t)
    ;; (setq org-agenda-todo-ignore-with-date t)
    ;; (setq org-agenda-todo-ignore-deadlines 'all)
    (setq org-agenda-todo-ignore-scheduled 'all)
    (setq org-agenda-todo-ignore-timestamp 'all)
    (setq org-hierarchical-todo-statistics nil)
    (setq org-startup-align-all-tables t)
    (setq org-agenda-restore-windows-after-quit nil)
    (setq org-use-property-inheritance nil)
    (setq org-agenda-sorting-strategy (quote ((agenda habit-down time-up priority-down category-keep)
                                              (todo tag-up priority-down todo-state-up category-keep)
                                              (tags priority-down category-keep)
                                              (search category-keep)))))
```
